frontend/src/main/components/RecommendationRequest/RecommendationRequestForm.js

```diff
@@ -0,0 +1,229 @@
+              type="text"
+              isInvalid={Boolean(errors.professorName)}
+              {...register("professorName", {
+                required: true,
+              })}
+            />
+            <Form.Control.Feedback type="invalid">
+              {errors.professorName && "Professor Name is required. "}
+            </Form.Control.Feedback>
+          </Form.Group>
+        </Col>
+      </Row>
+
+      <Row>
+        {/* Request Type */}
+        <Col>
+          <Form.Group className="mb-3">
+            <Form.Label htmlFor="requestType">Request Type</Form.Label>
+            <Form.Select
+              data-testid="RecommendationRequestForm-requestType"
+              id="requestType"
+              type="string"
+              isInvalid={Boolean(errors.requestType)}
+              {...register("requestType", {
+                required: "Request Type is required.",
+              })}
+            >
+              <option value="">-- Select an Option --</option>
+              <option value="CS Department BS/MS program">
+                CS Department BS/MS program
+              </option>
+              <option value="Scholarship or Fellowship">
+                Scholarship or Fellowship
+              </option>
+              <option value="MS program (other than CS Dept BS/MS)">
+                MS program (other than CS Dept BS/MS)
+              </option>
+              <option value="PhD program">PhD program</option>
+              <option value="Other">Other</option>
+            </Form.Select>
+            <Form.Control.Feedback type="invalid">
+              {errors.requestType?.message}
+            </Form.Control.Feedback>
+          </Form.Group>
+        </Col>
+
+        {/* Details */}
+        <Col>
+          <Form.Group className="mb-3">
+            <Form.Label htmlFor="details">Details</Form.Label>
+            <Form.Control
+              data-testid="RecommendationRequestForm-details"
+              id="details"
+              type="text"
+              placeholder="Enter additional details (optional)"
+              isInvalid={Boolean(errors.details)}
+              {...register("details")}
+            />
+          </Form.Group>
+        </Col>
+      </Row>
+
+      {/* Submission Date */}
+      <Row>
+        <Col>
+          <Form.Group className="mb-3">
+            <Form.Label htmlFor="submissionDate">Submission Date</Form.Label>
+            <Form.Control
+              data-testid="RecommendationRequestForm-submissionDate"
+              id="submissionDate"
+              type="datetime-local"
+              isInvalid={Boolean(errors.submissionDate)}
+              {...register("submissionDate", {
+                required: "Submission Date is required.",
+              })}
+            />
+            <Form.Control.Feedback type="invalid">
+              {errors.submissionDate?.message}
+            </Form.Control.Feedback>
+          </Form.Group>
+        </Col>
```

Comment:
No.   The submission date is calculated automatically in the backend.  The end user doesn't calculate it.